### PR TITLE
:bug: Fix flaky wait in CRS test

### PR DESCRIPTION
**Pull request by killianmuldoon**
I've seen a bunch of flakiness in this test - both locally and in CI. I think the issue is waiting for the APIServer to create the service which sometimes doesn't happen quickly. I've updated the test to just create the service directly.

/area testing

### CODE REVIEW

File: exp/addons/internal/controllers/clusterresourceset_controller_test.go [hunk 1]
```diff
@@ -978,28 +971,27 @@ metadata:
...
+		clusterResourceSetBindingKey := client.ObjectKey{Namespace: testCluster.Namespace, Name: testCluster.Name}
 		g.Consistently(func() bool {
 			binding := &addonsv1.ClusterResourceSetBinding{}
 
 			err := env.Get(ctx, clusterResourceSetBindingKey, binding)
 			return apierrors.IsNotFound(err)
 		}, timeout).Should(BeTrue())
 
-		t.Log("Make sure Kubernetes API Server Service has been created")
+		t.Log("Create Kubernetes API Server Service has been created")
```
Comment (sbueringer) [inline exp/addons/internal/controllers/clusterresourceset_controller_test.go:982]: nit: text seems a bit off

**Comment by sbueringer** [comment]
@killianmuldoon I think v1.2 & v1.3 should be good as we just implemented this feature & test with v1.4?

**Comment by killianmuldoon** [comment]
Yeah - didn't check to see which branches the test was actually on

**Pull request merged**